data/data/bootstrap/files/usr/local/bin/bootstrap-pivot.sh.template

```diff
@@ -42,25 +42,67 @@ if [ ! -f /opt/openshift/.pivot-done ]; then
+  if grep -q coreos.liveiso= /proc/cmdline; then
+    mount -t tmpfs -o size=50% none /var/mnt/
+    rsync -aHAXx "$mnt/" /var/mnt/
+    mount -t overlay overlay -o lowerdir=/usr:/var/mnt/usr /usr
+    rsync -rlt --ignore-existing /var/mnt/etc/ /etc/
 
-  systemctl start crio.service
-  # No reboot necessary because SNO setup will reboot system
-{{ else }}
-  pushd "${mnt}/bootstrap"
-    # shellcheck disable=SC1091
-    . ./pre-pivot.sh
-  popd
-{{ end -}}
+    # Agent-based Installer will launch a ephemeral control plane at the rendezvous host which will create and publish
+    # Ignition configs for the other master nodes. These Ignition configs must match what the in-cluster control plane
+    # would generate else machine config operator will fail [0]. Because the rendezvous host is booted with a FCOS Live
+    # ISO without any OKD/FCOS related changes, we have to copy the manifests from OKD Machine OS manually to the
+    # bootstrap manifests folder of the rendezvous host.
+    # [0] https://access.redhat.com/solutions/4970731
+    mkdir -p /var/opt/openshift/manifests
+    cp -av /var/mnt/manifests/*.* /var/opt/openshift/manifests/
+
+    # Load new systemd unit files and configuration such as crio.service after mounting the content of OKD/FCOS Machine
+    # OS over /usr and copying new files to /etc
+    systemctl daemon-reload
+
+    # On OKD/FCOS prior to commit e859a66 [0] systemd-resolved is used by default and NetworkManager's DNS handling is
+    # disabled. In this case, CoreDNS fails to listen to 127.0.0.53:53 when Agent-based Installer boots its the
+    # rendezvous host with a Fedora CoreOS bootimage because by default FCOS' systemd-resolved already listens to this
+    # port. OKD/FCOS disables resolved's stub listener [1] but the resolved must be restarted for this setting to take
+    # effect.
+    # On OKD/FCOS since commit e859a66 [0] systemd-resolved is disabled by default and NetworkManager's DNS handling is
+    # used. However, the bootimage is vanilla FCOS and thus uses systemd-resolved by default. The latter has to be
+    # disabled after rebasing to OKD Machine OS and NetworkManager as well as the service to fix /etc/resolv.conf have
+    # to be started.
+    # [0] https://github.com/openshift/okd-machine-os/commit/e859a6643330596a8a282aeb4bf853763a2d219e#diff-808ba069aeee05cbeb08aa7b8b5b4f6feb8aefe15ea9737339d07f8d7bf5d74a
+    # [1] https://github.com/openshift/okd-machine-os/blob/28dec35d60ea07069366b22ebdcb296d429b15e9/overlay.d/99okd/etc/systemd/resolved.conf.d/okd-no-dns-stub.conf
+    if [ -e /etc/systemd/resolved.conf.d/okd-no-dns-stub.conf ]; then
+        systemctl restart systemd-resolved.service
+    else
+        systemctl disable --now systemd-resolved.service
+    fi
+
+    if systemctl list-unit-files -q fix-resolvconf.service >/dev/null; then
+        systemctl start fix-resolvconf.service
+        systemctl restart NetworkManager.service
```

Comment:
we have to manually start fix-resolvconf.service because neither `daemon-reload` nor `preset-all` will start or stop any services.

Reply:
Through daemon-reload and preset-all we will get the services (re-)configured and enabled.
In particular, fix-resolvconf.service will be started `before` NetworkManager.service starts. I'm worried that executing it while NM is running (before launching the restart, i.e., before NM stops) can give an uncertain behavior like filling the empty file with content given by the previous NM configuration and then NM doesn't update it again as it finds it in a good shape. See https://github.com/openshift/okd-machine-os/blob/master/overlay.d/99okd/usr/lib/systemd/system/fix-resolvconf.service#L4

Reply:
Note that this whole script gets executed *only* at a live iso during cluster install. There will not be a reboot which would cause the new services to be launched. If we do not start `fix-resolvconf.service` manually, nobody else will. Restarting `NetworkManager.service` will not start `fix-resolvconf.service`.

Reply:
@JM1 you're right. We should have added WantedBy=NetworkManager.service maybe to make what I want. In this case, can it be possible to run 
```
systemctl stop NetworkManager.service
systemctl start fix-resolvconf.service
systemctl start NetworkManager.service
```
to run exactly like in the case of the rebased system?

Reply:
Done!